packages/lib/pod/src/pod.ts

```diff
@@ -0,0 +1,166 @@
+import { requireDefinedParameter } from "@pcd/util";
+import JSONBig from "json-bigint";
+import { PODContent } from "./podContent";
+import { signPODRoot, verifyPODRootSignature } from "./podCrypto";
+import { PODEntries } from "./podTypes";
+import { checkPublicKeyFormat, checkSignatureFormat } from "./podUtil";
+
+/**
+ * Class encapsulating a signed POD with functions for common use cases.
+ * POD instances are immutable (within the limits of TypeScript), but derived
+ * data (such as the Merkle tree of entries) is calculated lazily as it is
+ * needed.
+ *
+ * A POD is made up of `PODEntries`, built into a Merkle tree (in sorted order)
+ * to produce a root hash called the Content ID, which is then signed.  To
+ * create a POD, use one of the static factory methods of this class.
+ *
+ * TODO(artwyman): Pointer to more detailed documention elsewhere.
+ *
+ * Most features depending on the POD entries but not the signature are
+ * provided by a PODContent instance available via `pod.content`.
+ */
+export class POD {
+  private _content: PODContent;
+  private _signature: string;
+  private _signerPublicKey: string;
+
+  private constructor(
+    content: PODContent,
+    signature: string,
+    signerPublicKey: string
+  ) {
+    this._content = content;
+    this._signature = signature;
+    this._signerPublicKey = signerPublicKey;
+  }
+
+  /**
+   * This POD's data as a PODContent object.
+   */
+  public get content(): PODContent {
+    return this._content;
+  }
+
+  /**
+   * The content ID (root hash) of this POD.  PODs containing the same data
+   * will have the same content ID.
+   */
```

Comment:
objects which have 1 entry each - one cryptographic and one int, both equal to 1, would result in the same content id, right? potentially worth pointing out.